models/dataset_test.go

```diff
@@ -135,6 +142,62 @@ func TestCreateVersion(t *testing.T) {
 	})
 }
 
+func TestValidateDataset(t *testing.T) {
+	t.Parallel()
+
+	Convey("Unsuccessful validation (false) returned", t, func() {
+
+		Convey("when dataset.URI is unable to be parsed into url format", func() {
+			dataset := createDataset()
+			dataset.URI = ":foo"
+			fmt.Println(dataset.URI)
+			validationErr := ValidateDataset(testContext, &dataset)
+			So(validationErr, ShouldNotBeNil)
+			So(validationErr.Error(), ShouldResemble, errors.New("invalid fields: [URI]").Error())
+		})
+
+		Convey("when dataset.URI does not contain 'datasets' path", func() {
+			dataset := createDataset()
+			dataset.URI = "/123"
+			valid := ValidateDataset(testContext, &dataset)
+			So(valid, ShouldNotBeNil)
+			So(valid.Error(), ShouldResemble, errors.New("invalid fields: [URI]").Error())
+		})
+
+		Convey("when dataset.URI does not contain 'id' path", func() {
+			dataset := createDataset()
+			dataset.URI = "http://localhost:22000/datasets"
+			valid := ValidateDataset(testContext, &dataset)
+			So(valid, ShouldNotBeNil)
+			So(valid.Error(), ShouldResemble, errors.New("invalid fields: [URI]").Error())
+		})
+
+	})
+
+	Convey("Successful validation (true) returned", t, func() {
+
+		Convey("when dataset.URI contains its path in appropriate url format ", func() {
+			dataset := createDataset()
+			dataset.ID = "123"
+			dataset.URI = "http://localhost:22000/datasets/123/breadcrumbs"
+			valid := ValidateDataset(testContext, &dataset)
+			So(valid, ShouldBeNil)
+		})
+	})
+
+	Convey("Successful validation (true) returned", t, func() {
+
+		Convey("when dataset.URI contains whitespace it should not return an error ", func() {
+			dataset := createDataset()
+			dataset.ID = "123"
+			dataset.URI = "  http://localhost:22000/datasets/123/breadcrumbs  "
+			valid := ValidateDataset(testContext, &dataset)
+			So(valid, ShouldBeNil)
+		})
```

Comment:
I think we should check that the dataset.URI has been correctly trimmed here. Can you add something like this, please?:
```
So(dataset.URI, ShouldEqual, "http://localhost:22000/datasets/123/breadcrumbs")
```